src/components/map/Map.jsx

```diff
@@ -1,42 +1,40 @@
-import React from 'react';
-import { compose, withProps, withStateHandlers, lifecycle } from 'recompose';
+import React, { Component } from 'react';
+import { compose, withProps, lifecycle } from 'recompose';
 import { withScriptjs, withGoogleMap, GoogleMap } from 'react-google-maps';
 import LocationMarker from './LocationMarker';
 import config from '../../config';
+import debounce from 'lodash/debounce';
 
-const Map = compose(
+const onBoundsChangedDebouncePeriod = 500;
+const MyMap = compose(
   withProps({
     googleMapURL: config.googleMaps,
     loadingElement: <div style={{ height: '100%' }} />,
     containerElement: <div style={{ height: '100%' }} />,
     mapElement: <div style={{ height: '100%' }} />,
   }),
-  withStateHandlers(
-    () => ({
-      openLocationId: null,
-    }),
-    {
-      onToggleMarkerInfo: ({ openLocationId }) => toggledLocationId => ({
-        openLocationId: openLocationId === toggledLocationId ? null : toggledLocationId,
-      }),
-    },
-  ),
   lifecycle({
-    componentWillMount() {
+    componentWillMount(){
       let mapRef;
 
       this.setState({
         onMapMounted: (ref) => {
           mapRef = ref;
         },
-        onCenterChanged: () => {
-          const center = mapRef.getCenter();
-          if (this.props.onCenterChanged) {
-            this.props.onCenterChanged({ lat: center.lat(), lng: center.lng() });
-          }
-        },
+        onBoundsChanged: debounce(() => {
+          const bounds = mapRef.getBounds()
+          const center = mapRef.getCenter()
+          if(!bounds || !center) return;
+          const radius = window.google.maps.geometry.spherical.computeDistanceBetween(
+            center,
+            {
+              lat: () => bounds.f.b, lng: () => bounds.b.b
```

Comment:
Not sure where `f.b` and `b.b` come from. I was sure when I looked at the bounds, they had normally-named lat and long properties...

Reply:
Not sure.